tracer/src/Datadog.Trace/Debugger/Instrumentation/AsyncMethodDebuggerInvoker.SingleProbe.cs

```diff
@@ -234,6 +236,24 @@ public static void LogLocal<TLocal>(ref TLocal local, int index, ref AsyncMethod
             asyncState.SnapshotCreator.StartSampling();
         }
 
+        private static bool ProcessNotHoistedMoveNextLocal<TLocal>(ref TLocal local, AsyncMethodDebuggerState asyncState, string localName)
+        {
+            bool isSucceeded;
+            if (TypeExtensions.IsDefaultValue(ref local))
```

Comment:
I wonder if checking the default value is the correct behavior, what if the local that is not hoisted simply didn't have the value?
I'm not sure how we want to tackle it, but treating all those locals as unreachable could be misleading.

Reply:
You right but other ways to check it are more complicated and we still don't know if it worth it. For now we decided to miss default values from the snapshot completely than display something that might be incorrect.